test/cancelable_operation_test.dart

```diff
@@ -460,8 +460,10 @@ void main() {
         var operation = runThen();
         var workCompleter = Completer<int>();
         originalCompleter.complete(workCompleter.future);
-        originalCompleter.operation.cancel();
+        var cancelation = originalCompleter.operation.cancel();
+        expect(originalCompleter.isCanceled, true);
```

Comment:
How does `isCanceled` become true synchronously here while we had to add the `await` above?

Reply:
The `isCanceled` of the operation you cancel gets set immediately.
Then the cancellation propagates asynchronously to `.then` listeners, which is what the other test checks (`operation.isCanceled` vs `originalCompleter.operation.isCanceled`.)